ocaml/xapi/message_forwarding.ml

```diff
@@ -3295,24 +3295,45 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
 
     let destroy ~__context ~sr =
       info "SR.destroy: SR = '%s'" (sr_uuid ~__context sr);
-      let local_fn = Local.SR.destroy ~sr in
+      let local_destroy = Local.SR.destroy ~sr in
+      let local_forget = Local.SR.forget ~sr in
       with_sr_marked ~__context ~sr ~doc:"SR.destroy" ~op:`destroy
         (fun () ->
-          Xapi_sr.assert_all_pbds_unplugged ~__context ~sr;
+          let pbd_state = Xapi_sr.assert_all_pbds_unplugged ~__context ~sr in
           Xapi_sr.assert_sr_not_indestructible ~__context ~sr;
           Xapi_sr.assert_sr_not_local_cache ~__context ~sr;
 
-          forward_sr_op ~consider_unplugged_pbds:true ~local_fn ~__context ~self:sr
-            (fun session_id rpc -> Client.SR.destroy rpc session_id sr))
+          forward_sr_op ~consider_unplugged_pbds:true ~local_fn:local_destroy ~__context ~self:sr
+            (fun session_id rpc -> Client.SR.destroy rpc session_id sr);
+
+          begin
+            match pbd_state with
+            | `No_pbds            -> (* no SR metrics to remove *) ()
+            | `All_pbds_unplugged ->
```

Comment:
Do we really need to distinguish the no-PBD case here? Can we not just let the `forward_sr_all_op` be a no-op in that case?

In this light, perhaps it is better if `choose_pbds_for_sr` does not raise errors, but delegates that to `choose_pbd_for_sr`.